core/src/trezor/wire/thp/crypto.py

```
@@ -157,8 +160,9 @@ def handle_th2_crypto(
         host_static_public_key = memoryview(encrypted_host_static_public_key)[
             :PUBKEY_LENGTH
         ]
-        tag = aes_ctx.finish()
-        if not utils.consteq(tag, encrypted_host_static_public_key[-16:]):
+        try:
+            aes_ctx.finish(encrypted_host_static_public_key[-16:])
+        except RuntimeError:
             raise ThpDecryptionError()
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)